implementation/leprechaun/src/shared/api/validation.service.ts

```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@nestjs/common';
+import { validate } from 'class-validator';
+import { SchemaMismatchException } from './http-exceptions';
+
+@Injectable()
+export class ValidationService {
+  // We could possible redo this as validation-pipe, CreateObjectValidationPipe and PatchObjectValidationPipe
+  async validate<T extends object>(obj: T, update: boolean = false): Promise<boolean> {
```

Comment:
**question:** Is there any situation where this would return a `false`? If not, the signature can be changed to `Promise<true>`, and if that's the case, it might as well be `Promise<void>` since the result is useless.